tests/system.py

```diff
@@ -434,7 +434,9 @@ def test_delete_dataset_delete_contents_true(self):
         dataset_id = _make_dataset_id("delete_table_true")
         project = Config.CLIENT.project
         dataset_ref = bigquery.DatasetReference(project, dataset_id)
-        dataset = retry_403(Config.CLIENT.create_dataset)(Dataset(dataset_ref))
+        dataset = retry_403(Config.CLIENT.create_dataset)(
```

Comment:
We can also check dataset exist and then delete it before create, but think that's incorrect way to test, so we can use `exist_ok` as it doesn't matter in this test if it's already exist due to some reason.

Should i have to use retry here with 403 and 409 errors?

Reply:
I think a better fix is to avoid sharing the dataset ID between this method and the one above:  the issue seems to be eventual consistency.

Reply:
Yes, we are appending UUID but still it's possible so changed the dataset id.